docs/compiler-options.md

```diff
@@ -13,6 +13,7 @@
 | `-Osimplify-main` `-Ono-simplify-main` | Disabled | [simplify-main](#simplify-main) |
 | `-Omerge` `-Ono-merge` | Disabled | [definition-merging](#definition-merging) |
 | `-Oinline` `-Ono-inline` | Disabled | [inline](#inline) |
+| `-e` `--entrypoint` | `Main \| main` | Use another definition as entrypoint, rather than `main` or `Main`. |
```

Comment:
I think it's best to expand this into it's own section so that we can explain the compiled output difference between having a custom entry point and not having one.